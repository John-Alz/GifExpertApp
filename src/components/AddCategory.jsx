import PropTypes from "prop-types";
import { useState } from "react";

export function AddCategory({ onNewCategory }) {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = (e) => {
        setInputValue(e.target.value)
    }

    const onSubmit = (e) => {
        e.preventDefault();
        const newInputValue = inputValue.trim().toLowerCase();
        if (newInputValue.length <= 1) return; // Debe ser mayor de 2 caracteres el input
        // // onAddCategories(cat => [...cat, inputValue]);
        onNewCategory(newInputValue);
        setInputValue('');
    }

    return (
        <form aria-label="form" onSubmit={onSubmit} >
            <input
                type="text"
                placeholder="Buscar Gifs"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}


AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}
