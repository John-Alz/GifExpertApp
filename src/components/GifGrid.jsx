
export default function GifGrid({ category }) {
    return (
        <>
            <h3>{category}</h3>
        </>
    )
}
