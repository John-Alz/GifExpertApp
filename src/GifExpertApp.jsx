import { useState } from 'react'
import { AddCategory, GifGrid } from './components';

export default function GifExpertApp() {

    const [categories, setCategories] = useState(['Dragon ball']);

    const onAddCategory = (keyWord) => {

        if (categories.includes(keyWord)) return;

        setCategories([
            keyWord,
            ...categories
        ]);

        // setCategories(cat => [...cat, 'Valorant']) <== Importante en <AddCategory/> comentado
    };

    const onDeleteCategory = (keyWordDelete) => {
        const newArr = categories.filter(word => word !== keyWordDelete);
        setCategories(newArr)
    }

    return (
        <>
            <h1>GifExpertApp</h1>

            <AddCategory onNewCategory={onAddCategory} />

            {
                categories.length > 0 ? categories.map((category) =>
                (
                    <GifGrid key={category} category={category} categories={categories} onDeleteCategory={onDeleteCategory} />
                )
                ) : <h2 className='empty-grid'>Busca tus gifs favoritos!!!</h2>
            }
        </>
    )
}
