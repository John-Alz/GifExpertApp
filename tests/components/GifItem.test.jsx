import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components";

describe('Pruebas para el componente <GifItem/>', () => {

    const title = 'Saitama';
    const url = 'https://one-punch.com/saitama.jpg';

    test('Debe hacer match con el Snapshot del componente', () => {

        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();

    });

    test('Debe mostar la iamgen con el URL y ALT indicado', () => {

        render(<GifItem title={title} url={url} />); // Sujeto de pruebas
        // screen.debug();
        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);

        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);

    });


    test('Debe de mostar el titulo en el componente', () => {

        render(<GifItem title={title} url={url} />); //Sujeto de pruebas
        expect(screen.getByText(title)).toBeTruthy();

    })




});