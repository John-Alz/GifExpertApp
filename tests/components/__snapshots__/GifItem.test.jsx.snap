// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para el componente <GifItem/> Debe hacer match con el Snapshot del componente 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
